redi-connect-front/src/pages/front/signup/SignUp.tsx

```diff
@@ -0,0 +1,322 @@
+import React, { useState } from "react";
+import * as Yup from "yup";
+
+import { Formik, FormikValues, FormikActions } from "formik";
+import Grid from "@material-ui/core/Grid";
+import {
+  TextField,
+  Select,
+  MenuItem,
+  FormControl,
+  InputLabel,
+} from "@material-ui/core";
+import omit from "lodash/omit";
+
+import { Button, FormControlLabel, Checkbox } from "@material-ui/core";
+
+import { signUp } from "../../../services/api/api";
+import { RedProfile } from "../../../types/RedProfile";
+import { history } from "../../../services/history/history";
+
+import { courses } from "../../../config/config";
+
+export const validationSchema = Yup.object({
+  firstName: Yup.string()
+    .required()
+    .max(255),
+  lastName: Yup.string()
+    .required()
+    .max(255),
+  username: Yup.string()
+    .email()
+    .label("Email")
+    .max(255),
+  password: Yup.string()
+    .min(8, "Password must contain at least 8 characters")
+    .required("Enter your password")
+    .label("Password"),
+  passwordConfirm: Yup.string()
+    .required("Confirm your password")
+    .oneOf([Yup.ref("password")], "Password does not match"),
+  agreesWithCodeOfConduct: Yup.boolean()
+    .required()
+    .oneOf([true]),
+  gaveGdprConsentAt: Yup.string()
+    .required()
+    .label("Data usage consent"),
+  mentee_currentlyEnrolledInCourse: Yup.string().when("formType", {
+    is: "public-sign-up-mentee-pending-review",
+    then: Yup.string()
+      .oneOf(courses.map(level => level.id))
+      .label("Currently enrolled in course"),
+  }),
+});
+
+export type SignUpFormType =
+  | "mentor"
+  | "mentee"
+  | "public-sign-up-mentor-pending-review"
+  | "public-sign-up-mentee-pending-review";
+
+export interface SignUpFormValues {
+  formType: SignUpFormType;
+  gaveGdprConsentAt: string;
+  username: string;
+  password: string;
+  passwordConfirm: string;
+  firstName: string;
+  lastName: string;
+  agreesWithCodeOfConduct: boolean;
+  mentee_currentlyEnrolledInCourse: string;
+}
+
+const initialValues: SignUpFormValues = {
+  formType: "mentee",
+  gaveGdprConsentAt: "",
+  username: "",
+  password: "",
+  passwordConfirm: "",
+  firstName: "",
+  lastName: "",
+  agreesWithCodeOfConduct: false,
+  mentee_currentlyEnrolledInCourse: "",
+};
+
+export const buildSignUpForm = (
+  type: SignUpFormType
+): Function => (): React.ReactFragment => {
+  const [submitError, setSubmitError] = useState(false);
+  const submitForm = async (
+    values: FormikValues,
+    actions: FormikActions<SignUpFormValues>
+  ) => {
+    setSubmitError(false);
+    const profile = values as Partial<RedProfile>;
+    // TODO: this needs to be done in a smarter way, like iterating over the RedProfile definition or something
+    const cleanProfile: Partial<RedProfile> = omit(profile, [
+      "password",
+      "passwordConfirm",
+      "formType",
+      "agreesWithCodeOfConduct",
+    ]);
+    cleanProfile.userType = type;
+    cleanProfile.userActivated = false;
```

Comment:
These properties (`userActivated` and `signupSource`) should be set (and enforced) by the back-end. Can you create a ticket, @ezonai ?

Reply:
done